src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1263,11 +1309,15 @@ private Pager.State ApplyPagesToDataFileFromScratch(ApplyLogsToDataFileState sta
                         {
                             for (int i = 0; i < pages.Length; i++)
                             {
-                                Debug.Assert(pages[i].PageNumber + pages[i].GetNumberOfPages() <= dataPagerState.NumberOfAllocatedPages,
+                                int numberOfPages = pages[i].GetNumberOfPages();
+
+                                Debug.Assert(pages[i].PageNumber + numberOfPages <= dataPagerState.NumberOfAllocatedPages,
                                     "pages[i].PageNumber + pages[i].GetNumberOfPagesUpdateStateOnCommit() <= dataPagerState.NumberOfAllocatedPages");
```

Comment:
Consider that now we can use `PortableExceptions.ThrowOnDebug` in cases where the `throw` itself makes sense instead of an assert.